lib/fauna/page.rb

```diff
@@ -221,6 +221,38 @@ def reverse_each
       end
     end
 
+    ##
+    # Returns the flattened contents of the set as an array.
+    #
+    # Ideal for when you need the full contents of a set with a known small size. If you need to iterate over a set
+    # of large or unknown size, it is recommended to use +each+ instead.
+    #
+    # The set is paged in the +after+ direction.
+    def set_contents
+      each.flat_map { |x| x }
+    end
+
+    ##
+    # Iterates over the entire set, applying the configured map and discarding the result.
+    #
+    # Ideal for performing a +foreach+ over an entire set (like deleting all instances in a set). The set is iterated in
+    # the +after+ direction.
+    #
+    # Example of deleting every instance in a set:
+    #
+    #   page.apply_map! { |page_q| foreach(page_q) { |ref| delete ref } }
+    def apply_map!(&block)
```

Comment:
Here's a semantics question: What if `with_map` took a fauna lambda and passed that to a map expr, and applied the map directly, and similarly `foreach!` or `with_foreach` could do the same.

It looks like it's possible for users to break the internal logic if with_map or apply_map! do not actually return a page result.